### PR TITLE
single-transaction deposit of native token to legacy vault

### PR DESCRIPTION
**Pull request by xiaozhou**

### CODE REVIEW

File: contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                _amount,
+                _mintChainId,
+                _mintAccount,
+                _nonce,
+                uint64(block.chainid)
+            )
+        );
+        IWETH(nativeWrap).deposit{value: _amount}();
+        IOriginalTokenVault(vault).deposit(nativeWrap, _amount, _mintChainId, _mintAccount, _nonce);
+        senders[depositId] = msg.sender;
```
Comment (Dominator008) [inline contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol:53]: Emit an event to indicate the "real" depositor?

File: contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        return depositId;
+    }
+
+    /**
+     * @notice Refund a failed native token deposit
```
Comment (Dominator008) [inline contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol:58]: Is this reserved for unusual circumstances like unsupported tokens?

File: contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import "../../interfaces/IWETH.sol";
+
+// To support single-transaction deposit of native gas token (e.g., ETH in ethereum)
+// to legacy vaults that do not have the depositNative function
+// Note that the depositor address shown at the desitnation chain will be this contract address
+contract NativeTokenVaultProxy {
+    address public immutable nativeWrap;
+    address public immutable vault;
+
+    mapping(bytes32 => address) public senders;
```
Comment (Dominator008) [inline contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol:18]: depositors?

File: contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol (hunk 2 above)
Comment (xiaozhou) [reply contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol:58]: some uncommon cases for refund, including "not enough to cover fees", "exceed supply cap", "unsupported tokens"

File: contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol (hunk 1 above)
Comment (xiaozhou) [reply contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol]: not really useful? Especially given that the real depositor is recorded on chain

Comment (Dominator008) [reply contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol]: For history monitoring?

Comment (xiaozhou) [reply contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol]: added

**Pull request closed without merging**